core/src/apps/bitcoin/sign_tx/__init__.py

```diff
@@ -2,41 +2,55 @@
 from trezor.messages.RequestType import TXFINISHED
 from trezor.messages.SignTx import SignTx
 from trezor.messages.TxAck import TxAck
-from trezor.messages.TxRequest import TxRequest
 
-from apps.common import coininfo, paths, seed
+from apps.common import coininfo, paths
 
-from ..keychain import with_keychain
-from . import bitcoin, helpers, layout, progress
+from ..keychain import get_keychain_for_coin
 
 if not utils.BITCOIN_ONLY:
     from . import bitcoinlike, decred, zcash
 
 if False:
-    from typing import Type, Union
+    from typing import Optional, Union
+    from trezor.messages.TxRequest import TxRequest
+    from apps.common.seed import Keychain
+    from ..authorization import CoinJoinAuthorization
 
 
 BITCOIN_NAMES = ("Bitcoin", "Regtest", "Testnet")
 
 
-@with_keychain
-async def sign_tx(
-    ctx: wire.Context, msg: SignTx, keychain: seed.Keychain, coin: coininfo.CoinInfo
+async def sign_tx(ctx: wire.Context, msg: SignTx) -> TxRequest:
+    keychain, coin = await get_keychain_for_coin(ctx, msg.coin_name)
+    return await sign_tx_impl(ctx, msg, keychain, coin)
+
+
+async def sign_tx_impl(
+    ctx: wire.Context,
+    msg: SignTx,
+    keychain: Keychain,
+    coin: coininfo.CoinInfo,
+    authorization: Optional[CoinJoinAuthorization] = None,
 ) -> TxRequest:
-    if not utils.BITCOIN_ONLY:
+    from . import approvers, bitcoin, helpers, layout, progress
+    from trezor.messages.TxRequest import TxRequest
```

Comment:
Something fishy is going on here. Looks like `bitcoin` and some other modules are getting unimported.